src/script/interpreter.cpp

```diff
@@ -1087,9 +1116,23 @@ class CTransactionSignatureSerializer {
  
             if (txTo.IsScVersion() )
             {
+                // Serialize CSW inputs
+                // In case of SIGHASH_ANYONECANPAY:
+                // * if Tx has Sc support version and nIn belongs to the CSW inputs - only the CSW input being signed is serialized
+                // * otherwise skip CSW inputs
+                // Otherwise - serialize all CSW inputs
+                unsigned int nCswInputs = fAnyoneCanPay ?
+                            (nIn < txTo.GetVin().size() ? 0 : 1) :
+                            txTo.GetVcswCcIn().size();
+                ::WriteCompactSize(s, nCswInputs);
+                for (unsigned int nCswInput = 0; nCswInput < nCswInputs; nCswInput++)
+                    SerializeCswInput(s, nCswInput, txTo, nType, nVersion);
```

Comment:
Why you have chosen to serialize cws inputs after outputs?
Why don't we process all inputs (ordinary and cws) first and all outputs after (ordinary and cc ones)?

Reply:
as decided AFK, keep sidechain related data together after regular data.